src/pages/Account/FirebaseEmailHandler/PasswordReset.tsx

```diff
@@ -0,0 +1,177 @@
+}
+
+export default class PasswordReset extends React.Component<Props> {
+  readonly state: State = {
+    isSubmitting: false,
+    hasError: false,
+    showPasswordResetForm: false,
+    showPasswordResetSuccess: false,
+    errorMessage: '',
+    successMessage: '',
+    password: '',
+  };
+
+  componentDidMount() {
+    const { oobCode } = this.props;
+
+    if (!oobCode) {
+      alert('Error, no oobCode defined.');
+    }
+
+    // @see
+    // https://firebase.google.com/docs/auth/custom-email-handler
+    this.handleResetPassword(oobCode);
+  }
+
+  handleFormChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const { name, value } = event.target;
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  // @see
+  // https://firebase.google.com/docs/auth/custom-email-handler
+  handleResetPassword = (actionCode: string) => {
+    // Verify the password reset code
+    // is valid.
+    auth
+      .verifyPasswordResetCode(actionCode)
+      .then(() => {
+        this.setState({ showPasswordResetForm: true });
+      })
+      .catch(() => {
+        this.setState({
+          hasError: true,
+          errorMessage: 'Invalid or expired action code. Please reset the password again.',
+        });
+      });
+  }
+
+  handlePasswordResetSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!this.props.oobCode) {
+      return this.setState({
+        hasError: true,
+        errorMessage: 'Invalid or expired action code. Please reset the password again.',
+      });
+    }
+
+    if (!this.state.password) {
+      return alert('Invalid password.');
+    }
+
+    // Save the new password.
+    auth
+      .confirmPasswordReset(this.props.oobCode, this.state.password)
+      .then(() => {
+        this.setState({ showPasswordResetForm: false, showPasswordResetSuccess: true });
+      }).catch((err: Error) => {
+        console.error(err);
+        this.setState({
+          hasError: true,
+          errorMessage: 'Password is too weak(at least 8 characters) or the password reset link has expired.',
+        });
+      });
+  }
+
+  renderError(): React.ReactNode {
+    return (
+      <Container tag={Fade} className={FIREBASE_EMAIL_HANDLER_CONTAINER_CLASSES}>
+        <h2 className={FIREBASE_EMAIL_HANDLER_TITLE_CLASSES}>Sorry, there was an error</h2>
+        <p className={FIREBASE_EMAIL_HANDLER_SUBTITLE_CLASSES}>{this.state.errorMessage}</p>
+        <Row>
+          <a target="_blank" href="https://support.beenest.com/">
+            <Button color="primary" className={FIREBASE_EMAIL_HANDLER_BUTTON}>Contact us for further help.</Button>
+          </a>
+        </Row>
+      </Container>
+    );
+  }
+
+  renderPasswordResetForm(): React.ReactNode {
+    return (
+      <Container tag={Fade} className={FIREBASE_EMAIL_HANDLER_CONTAINER_CLASSES}>
+        <Col md="6" lg="5">
+          <h2 className={FIREBASE_EMAIL_HANDLER_TITLE_CLASSES}>Reset Password</h2>
```

Comment:
can this use a default "title" class instead of a custom one just for firebase page?

Reply:
my thought was to try and not pollute the customStyles file with a bunch of classes because itll definitely be hard to manage all the styles in one file.

Reply:
i can move the `home.styled` file to another location and try and reuse things from there

Reply:
im suggesting not to use a custom class. 
Can it just be `<h2>` ?

Reply:
we'll have the same issue as we did for the home page where the title is good for desktop but too large for mobile, and if we just default `h` tags they have their own font-weight and size which will cause similar problems/not look good.  Front also slaps these custom classes on their text (this is where i got the combinations from)